src/Servant/Auth/Hmac.hs

```diff
@@ -1,6 +1,38 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+-- | Signing functions.
+
 module Servant.Auth.Hmac
-       ( foo
+       ( -- * Signing
+         sign
+       , signSHA256
+
+         -- * Secret key
+       , SecretKey (..)
        ) where
 
-foo :: String
-foo = "foo"
+import Crypto.Hash.Algorithms (SHA256)
+import Crypto.Hash.IO (HashAlgorithm)
+import Crypto.MAC.HMAC (HMAC (hmacGetDigest), hmac)
+import Data.ByteString (ByteString, pack)
+
+import qualified Data.ByteArray as BA (unpack)
+
+
+-- | The wraper for the secret key.
+newtype SecretKey = SecretKey
+    { unSecretKey :: ByteString
+    }
+
+-- | Compute the hashed message using the supplied hashing function.
+sign :: forall a . (HashAlgorithm a)
+     => SecretKey   -- ^ Secret key to use
+     -> ByteString  -- ^ Message to MAC
+     -> ByteString  -- ^ Hashed message
+sign (SecretKey sk) msg = pack $ BA.unpack $ hmacGetDigest $ hmac @_ @_ @a sk msg
```

Comment:
I think we can use [`convert`](https://hackage.haskell.org/package/memory-0.14.14/docs/Data-ByteArray.html#v:convert) function from `memory` package to convert `Digest` to `ByteString`.